sdk/storage/Azure.Storage.DataMovement/src/Shared/Errors.DataMovement.cs

```diff
@@ -128,5 +128,8 @@ public static ArgumentException NoResourceProviderFound(bool isSource, string pr
 
         public static ArgumentException UnexpectedPropertyType(string propertyName, params string[] expectedTypes)
             => new ArgumentException($"Unexpected property type encountered for storage resource property {propertyName}: {string.Join(",", (string[])expectedTypes)}");
+
+        public static InvalidOperationException CheckpointerDisabled()
+            => new InvalidOperationException("The transfer checkpointer is disabled.");
```

Comment:
Since this is being thrown when people call Resume (and if the checkpointer is disabled) would a more helpful error be, "Cannot perform {Insert API name here e.g. ResumeAllTransfers}. The transfer checkpointer is disabled" ?

Reply:
Sure, but I had to split into two errors. The checkpointer itself also can throw this error so I left that throwing the generic one but update the resume APIs to use the method specific one. In practice the generic error should not be hit, it's there just in case a call makes it to the checkpointer.

Reply:
You can add a parameter to pass in the API name that is called and that is failing. That way it's just one error, but looks like two since it tells us where it's coming from

Reply:
I previously meant that there were some instances of this error where we didn't want to return the method name to the user in the error since they were internal checkpointer methods, so I needed one that took a method name and one that was generic. However, I changed it to all instances providing their method name.